### ISSUE TITLE
How to call analyzeResult function？

### ISSUE THREAD

**Issue by YUESAI** [ma-compbio/PEP#1]
How to call analyzeResult or analyzeResul_temp functions in mainPEP.py?
What is the content of the incoming parameter model？

**Comment by yangymargaret** [comment]
Hi YUESAI,
Thank you for your questions.  
i.  analyzeResult and analyzeResult_temp were used in our old versions to test the performance of PEP-Word. We replaced them with score_function in the new version but retained them in the code.

The arguments of analyzeResult(data,model,DataVecs,threshold) and analyzeResult_temp(data,model,DataVecs) are as follows. 

‘data’: sample information loaded by calling data = getData(type,cell), which contains three columns: label (true class label of the samples), predicted label (to be filled after model fitting and prediction), predicted probability (to be filled after model fitting and prediction).

‘model’: the trained estimator.

‘DataVecs’: feature vectors of the samples generated by PEP-Word.

‘threshold’: threshold used to determine predicted labels with predicted probability.

To use analyzeResult and analyzeResult_temp, please follow the example below. The example does not include cross validation. You could adjust it according to your own use. 

Example: 

In run_word beginning from Line 256 of mainPEP.py, we load sample label information and generate feature vectors by calling 

data = getData(type,cell) (Line 263)

dataDataVecs = np.load("./Datavecs/datavecs_"+str(cell)+"_"+str(type)+".npy") (Line 266)
x = sklearn.preprocessing.StandardScaler().fit_transform(dataDataVecs) # standardize the features
y = data[“label”] # true class label
y = np.asarray(y)

(the following code are added)

\# train the model
model = xgb.XGBClassifier(max_depth=10, learning_rate=0.1, n_estimators=1000, nthread=50)
model.fit(x,y,sample_weight=data[“weight”])

\# use analyzeResult and analyzeResult_temp to evaluate the performance
threshold = 0.5 # the threshold can be adjusted using inner-round cross validation, as  performed by threshold_estimate() or threshold_estimate_cv() in parametered_cv() (beginning from Line 190 in mainPEP.py)

\# performance evaluation using adjusted thresholds
analyzeResult(data,x,model,threshold)

\# performance evaluation without using adjusted thresholds
analyzeResult_temp(data,x,model)

ii. Could you please specify the incoming parameter model you refer to? If it is about the input parameters, please refer to the function parse_args() on line 6 of PEP.py and the readme.md, where we give descriptions about the arguments.

Please let me know if you have any other questions. Many thanks!

**Comment by YUESAI** [comment]
Dear yangymargaret:
Thank you for your reply.

I'm sorry I did not say it clearly enough. The incoming parameter model I want to know is the parameter of analyzeResult .You have already answered in i..

In addition, do you have the code of this paper (Predicting Enhancer-Promoter Interaction from Genomic Sequence with Deep Neural Networks)? I am very interested in it. If you have, could you share it with me?

This is my first question on github. I am very touched by your detailed and enthusiastic response. Your answer solved my problem perfectly.

Many thanks!

**Comment by yangymargaret** [comment]
Dear YUESAI,
You're welcome and thank you again for your question.

The code of SPEID  (Predicting Enhancer-Promoter Interaction from Genomic Sequence with Deep Neural Networks) is available at https://github.com/ma-compbio/SPEID. Please follow the readme to use the code. 

Please let me know if you have any other questions. Many thanks!
